qa/tasks/keycloak.py

```diff
@@ -0,0 +1,351 @@
+"""
+Deploy and configure Keycloak for Teuthology
+"""
+from io import BytesIO
+from io import StringIO
+from configobj import ConfigObj
+import base64
+import argparse
+import contextlib
+import logging
+import os
+import random
+import six
+import string
+import subprocess
+import json
+import sys
+from pathlib import Path
+
+from collections import OrderedDict
+from itertools import chain
+
+from teuthology import misc as teuthology
+from teuthology import contextutil
+from teuthology.config import config as teuth_config
+from teuthology.orchestra import run
+from teuthology.packaging import install_package
+from teuthology.packaging import remove_package
+from teuthology.exceptions import ConfigError
+
+log = logging.getLogger(__name__)
+
+def get_keycloak_dir(ctx):
+    global version
```

Comment:
Since `global` in a function is used for writing to variables in a function that are global. I'd recommend you initialize the variable at the top first (ex:  under `log`) to `None`. Also I would change the name of `version` to `keycloak_version` so that the variable is more specific.

Reply:
I already tried shifting the ``global`` below log but whenever I change/assign that variable inside some function then it used to give me some python related error so I searched for it and then made this change.

Okay, I'll change the variable name to ``keycloak_version``